clients/src/test/java/org/apache/kafka/common/network/NioEchoServer.java

```diff
@@ -226,9 +228,17 @@ public void run() {
                 selector.poll(100);
                 synchronized (newChannels) {
                     for (SocketChannel socketChannel : newChannels) {
-                        String id = id(socketChannel);
-                        selector.register(id, socketChannel);
-                        socketChannels.add(socketChannel);
+                        String id = id();
+                        // This try-catch block prevents the NioEchoServer from crashing in a rare condition
+                        // when attempting to register new connections where a connection is already registered.
+                        // Without this, the test fails because the connection unexpectedly becomes idle and expires, leading to a timeout.
+                        try {
+                            selector.register(id, socketChannel);
+                            socketChannels.add(socketChannel);
+                        } catch (IllegalStateException e) {
```

Comment:
Do we need this cache after we use the unique id?

Reply:
Since we have loop tested the new id generator, I think it is solid and reliable, I will remove the `try-catch`